common/app/model/dotcomrendering/pageElements/PageElement.scala

```diff
@@ -181,9 +189,19 @@ object PageElement {
       }).toList
 
       case Embed => extractEmbed(element).toList
-
-      case Contentatom => element.contentAtomTypeData.map(d => ContentAtomBlockElement(d.atomId)).toList
-
+      case Contentatom => (extractAtom match {
+        case Some(mediaAtom: MediaAtom) =>
+          mediaAtom.activeAssets.headOption.map(asset => {
+            YoutubeBlockElement(
+              mediaAtom.id, //CAPI ID
+              asset.id,
+              mediaAtom.channelId, //Channel ID
+              mediaAtom.title //Caption
+            )
+          })
+        case Some(atom) => Some(ContentAtomBlockElement(atom.id))
```

Comment:
Does this mean all content atoms are now supported? Or is there more work to do here?

Reply:
No unfortunately, looking at it, ContentAtomBlockElement is just defined as:

    case class ContentAtomBlockElement(atomId: String) extends PageElement

which doesn't provide enough for DCR to do anything with it.

Reply:
Cool, thanks for clarifying. A place to build on, at least.